.github/workflows/rust.yml

```diff
@@ -79,28 +82,63 @@ jobs:
         uses: ./.github/actions/setup-builder
         with:
           rust-version: stable
-      - name: Check datafusion without default features
-        # Some of the test binaries require the parquet feature still
-        #run: cargo check --all-targets --no-default-features -p datafusion
-        run: cargo check --profile ci --no-default-features -p datafusion
-
-      - name: Check datafusion-common without default features
+      - name: Check datafusion-common (no-default-features)
         run: cargo check --profile ci --all-targets --no-default-features -p datafusion-common
+        # Note: don't check other feature flags as datafusion-common is not typically used standalone
 
-      - name: Check datafusion-functions without default features
-        run: cargo check --profile ci --all-targets --no-default-features -p datafusion-functions
-
-      - name: Check datafusion-substrait without default features
+  # Check datafusion-substrait features
+  #
+  # Ensure via `cargo check` that the crate can be built with a
+  # subset of the features packages enabled.
+  linux-datafusion-substrait-features:
+    name: cargo check datafusion-substrait features
+    needs: linux-build-lib
+    runs-on: ubuntu-latest
+    container:
+      image: amd64/rust
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Rust toolchain
+        uses: ./.github/actions/setup-builder
+        with:
+          rust-version: stable
+      - name: Check datafusion-substrait (no-default-features)
         run: cargo check --profile ci --all-targets --no-default-features -p datafusion-substrait
+      - name: Check datafusion-substrait (physical)
```

Comment:
this adds coverage of compiling datafusion-substrait with the available feature flags